tools/xtask/src/toolchain.rs

```diff
@@ -230,3 +224,23 @@ pub(crate) fn init_for_build(abi_changes_ok: bool) -> anyhow::Result<()> {
     std::env::set_var("RUSTDOC", &get_rustdoc_path()?);
     Ok(())
 }
+
+fn generate_config_toml() -> anyhow::Result<()> {
+    let mut data = File::open("toolchain/src/config.toml")?;
+    let mut buf = String::new();
+    data.read_to_string(&mut buf)?;
+    let mut toml = buf.parse::<Document>()?;
+    let host_triple = guess_host_triple::guess_host_triple().unwrap();
+
+    let curdir = std::env::current_dir().unwrap();
+    let llvm_bin = curdir.join("toolchain/src/rust/build").join(host_triple).join("llvm/bin");
+    toml["build"]["target"].as_array_mut().unwrap().push(host_triple);
+    toml["target"]["x86_64-unknown-twizzler"]["cc"] = toml_edit::value(llvm_bin.join("clang").to_str().unwrap());
+    toml["target"]["x86_64-unknown-twizzler"]["cxx"] = toml_edit::value(llvm_bin.join("clang++").to_str().unwrap());
+    toml["target"]["x86_64-unknown-twizzler"]["linker"] = toml_edit::value(llvm_bin.join("clang").to_str().unwrap());
+    toml["target"]["x86_64-unknown-twizzler"]["ar"] = toml_edit::value(llvm_bin.join("llvm-ar").to_str().unwrap());
```

Comment:
It does not have to happen in this patch, but in the future we should nail down the default behavior and add `arch/machine` flags like the other build steps xtask has. That way we can build support for ARM.

Reply:
Yes, agreed!